packages/core/src/browser/progress-location-service.ts

```diff
@@ -67,13 +66,25 @@ export class ProgressLocationService implements ProgressClient {
         const show = !!progressSet.size;
         this.fireEvent(locationId, show);
     }
-    protected readonly fireEvent = throttle((locationId: string, show: boolean) => {
+
+    protected lastEvents = new Map<string, { time: number, value: boolean }>();
+    protected fireEvent(locationId: string, show: boolean): void {
+        const lastEvent = this.lastEvents.get(locationId);
+
+        // we don't want to fire same events too often
+        const shouldFire = !lastEvent || lastEvent.value !== show || (lastEvent.time - Date.now()) > 250;
```

Comment:
Why do we need to fire the same value?

Reply:
You mean at all? I mean this service is there to back the UI and the assumption is, that a widget maybe closed and reopened, which in turn would register again for the same location to be notified, so you want to get updates, right?

Reply:
From code it does not look like that. If a client is registered afterwards it will miss it the last event anyway. It would be better then to have API like:
```ts
export interface LocationProgress {
   message?: string;
   show: boolean;
} 
/** fetch current progress */
getProgress(locationId): LocationProgress | undefined;
/** listen for progress changes */
onProgress(locationId): Event<LocationProgress>;
```

A client should fetch first current progress and then start listening for incoming.

Reply:
We probably don't need `show` anymore then and can indicate no progress by `undefined` value.

Reply:
sounds good. changing...

Reply:
I've updated, but left the `show` indicator in place. `message` property was unused. It's unsupported for vscode as well, and doesn't play nice with this approach.

Reply:
`ProgressBar` still does not care about initial progress: https://github.com/eclipse-theia/theia/pull/7314#discussion_r390859692